### PR TITLE
Additional Comment features

### PR DESCRIPTION
**Pull request by CarolineHana**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -20,12 +20,14 @@
 public final class Comment {
 
   private final long id;
+  private final String username;
   private final String text;
   private final String time;
 
 
-  public Comment(long id, String text, String time) {
+  public Comment(long id, String username, String text, String time) {
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: userName.

Prefer camelcase for function arguments https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -36,22 +36,37 @@
...
     List<Comment> comments = new ArrayList<>();
+
+    int counter = 0;
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
+      String username = (String) entity.getProperty("username");
       String text = (String) entity.getProperty("text");
       String time = (String) entity.getProperty("time");
-      Comment comment = new Comment(id, text, time);
+      Comment comment = new Comment(id, username, text, time);
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: same here

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -64,24 +79,60 @@ public void doGet (HttpServletRequest request, HttpServletResponse response) thr
...
+     String[] nameErrorArr = new String[]{"invalid name input"};
+     String[] textErrorArr = new String[]{"invalid text input"};
+     String username = request.getParameter("name-input");
      String text = request.getParameter("text-input");
 
+    
+    if(username.isEmpty()){
+            Gson gson = new Gson();
+            response.setContentType("application/json");
+            response.getWriter().println(gson.toJson(nameErrorArr));
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: it looks like nameErrorArr isn't used anywhere.  You can just do

response.getWriter().println(gson.toJson(new String[]{"invalid name input"}));

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -64,24 +79,60 @@ public void doGet (HttpServletRequest request, HttpServletResponse response) thr
...
+            return;
+        }
+    if(text.isEmpty()){
+            Gson gson = new Gson();
+            response.setContentType("application/json");
+            response.getWriter().println(gson.toJson(textErrorArr));
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: same here

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -64,24 +79,60 @@ public void doGet (HttpServletRequest request, HttpServletResponse response) thr
...
 
-    response.sendRedirect("/main_ENG.html");
+      response.sendRedirect("/main_ENG.html");
+      
 
    }
+
+     private int geShownComments(HttpServletRequest request) {
+    // Get the input from the form.
+     String showAmtString = request.getParameter("showAmt");
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: showAmountText

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -64,24 +79,60 @@ public void doGet (HttpServletRequest request, HttpServletResponse response) thr
...
+
+    // Convert the input to an int.
+    int showAmt;
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: showAmount

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -64,24 +79,60 @@ public void doGet (HttpServletRequest request, HttpServletResponse response) thr
...
 
 
 
-    response.sendRedirect("/main_ENG.html");
+      response.sendRedirect("/main_ENG.html");
+      
 
    }
+
+     private int geShownComments(HttpServletRequest request) {
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: getAmount

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -36,22 +36,37 @@
...
       comments.add(comment);
-     
+
+      counter++;
+      if (counter == 0) break;
+
+          
+      
     }
+     int showAmt = geShownComments(request);
```
Comment (ciddu) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: What if comments is empty? Can you return early here?

**Review by ciddu — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 1 above)
Comment (CarolineHana) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: oh I never seen this before, thanks for linking it!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 8 above)
Comment (CarolineHana) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:62]: good point, important edge case!

Comment (CarolineHana) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: good point, I was sort of thinking that this edge case was covered by the previous line of code:
if (counter == 0) break;

Comment (CarolineHana) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: actually, I think your right if the user requests 0 comments it should return early, I'll fix that

**Review by ciddu — APPROVED** [review]
Disable eslint before submitting.

**Pull request merged**